### PR TITLE
Import 3d model

### PR DESCRIPTION
**Pull request by cb299792458**

### CODE REVIEW

File: frontend/src/components/MapDisplay.jsx [hunk 1]
```diff
@@ -27,19 +27,50 @@ const MapDisplay = ({gameMap, tileConfig}) => {
...
+
+        // add castle at end
+        const loader = new GLTFLoader();
+        loader.load('/castle_3d_model.glb', (gltf) => {
+            gltf.scene.scale.set(0.005,0.005,0.005);
+            gltf.scene.position.set(
+                gameMap.width-tileConfig.width/2 - center.x,
+                (0.5+gameMap.heightMap.at(-1).at(-1))*tileConfig.height,
+                gameMap.width-tileConfig.width/2 - center.z,
+            );
```
Comment (rockokw) [inline frontend/src/components/MapDisplay.jsx:55]: We can make such things more explicit in the map itself later on.

File: frontend/src/components/MapDisplay.jsx [hunk 2]
```diff
@@ -27,19 +27,50 @@ const MapDisplay = ({gameMap, tileConfig}) => {
...
+            gltf.scene.scale.set(0.1,0.1,0.1);
+            gltf.scene.position.set(
+                0+tileConfig.width/2 - center.x,
+                (0.5+gameMap.heightMap[0][0])*tileConfig.height,
+                0+tileConfig.width/2 - center.z,
+            );
+            gltf.scene.rotation.y = Math.PI/4;
+
+            scene.add(gltf.scene)
+        });
```
Comment (rockokw) [inline frontend/src/components/MapDisplay.jsx:73]: Let's handle the error case for this model too.

File: frontend/src/map/RandomMap.js [hunk 3]
```diff
@@ -5,11 +5,14 @@ class RandomMap extends GameMap {
     constructor() {
         const width = 12;
         const depth = 12;
-        const height = 5;
+        const height = 6;
         super(width, depth, height);
 
-        for (let x = -width/2; x < width/2; x++) {
-            for (let y = -depth/2; y < depth/2; y++) {
+        this.heightMap = [];
```
Comment (rockokw) [inline frontend/src/map/RandomMap.js:11]: I guess this is fine for now. I'm thinking we should add something to the `GameMap` interface, which can be used for defining things like entry/exit points and other interesting things on the map. I'll put it on my todo list.

**Review by rockokw — APPROVED** [review]

File: frontend/src/map/RandomMap.js (hunk 3 above)
Comment (cb299792458) [reply frontend/src/map/RandomMap.js:11]: Sounds like a plan. At some point, we should try to sit down and whiteboard more of the game flow.

Comment (rockokw) [reply frontend/src/map/RandomMap.js:11]: Agreed

**Pull request merged**